lib/baseFactory.js

```diff
@@ -158,7 +158,8 @@ class BaseFactory {
      * @param  {Object}     instance            The current instance of a factory
      * @param  {Object}     config
      * @param  {Datastore}  config.datastore    A datastore instance
-     * @param  {Datastore}  config.scmPlugin    A scm plugin instance
+     * @param  {ScmPlugin}  [config.scmPlugin]  A scm plugin instance
```

Comment:
Remove any optional params to baseFactory. These should be documented in the factories that require them.